receiver/k8sclusterreceiver/internal/hpa/hpa.go

```diff
@@ -91,7 +92,54 @@ func GetMetrics(hpa *autoscalingv2beta2.HorizontalPodAutoscaler) []*agentmetrics
 	}
 }
 
-func getResourceForHPA(hpa *autoscalingv2beta2.HorizontalPodAutoscaler) *resourcepb.Resource {
+func GetMetricsBeta(hpa *autoscalingv2beta2.HorizontalPodAutoscaler) []*agentmetricspb.ExportMetricsServiceRequest {
+	metrics := []*metricspb.Metric{
+		{
+			MetricDescriptor: hpaMaxReplicasMetric,
+			Timeseries: []*metricspb.TimeSeries{
+				utils.GetInt64TimeSeries(int64(hpa.Spec.MaxReplicas)),
+			},
+		},
+		{
+			MetricDescriptor: hpaMinReplicasMetric,
+			Timeseries: []*metricspb.TimeSeries{
+				utils.GetInt64TimeSeries(int64(*hpa.Spec.MinReplicas)),
+			},
+		},
+		{
+			MetricDescriptor: hpaCurrentReplicasMetric,
+			Timeseries: []*metricspb.TimeSeries{
+				utils.GetInt64TimeSeries(int64(hpa.Status.CurrentReplicas)),
+			},
+		},
+		{
+			MetricDescriptor: hpaDesiredReplicasMetric,
+			Timeseries: []*metricspb.TimeSeries{
+				utils.GetInt64TimeSeries(int64(hpa.Status.DesiredReplicas)),
+			},
+		},
+	}
+
+	return []*agentmetricspb.ExportMetricsServiceRequest{
+		{
+			Resource: getResourceForHPABeta(hpa),
+			Metrics:  metrics,
+		},
+	}
+}
+
+func getResourceForHPA(hpa *autoscalingv2.HorizontalPodAutoscaler) *resourcepb.Resource {
```

Comment:
Can the existing be updated to take an `ObjectMeta` so no repeated definition is required?

Reply:
good call, done